### PR TITLE
Giving action permissions

### PR DESCRIPTION
**Pull request by haraldsvik**
giving action permission to deploy by adding github token and info. 

[Worked!](https://github.com/statisticsnorway/microdata-tools/actions/runs/11031492401)

### CODE REVIEW

File: .github/workflows/docs-deploy.yml [hunk 1]
```diff
@@ -22,5 +20,13 @@ jobs:
         run: |
           pip install mkdocs-material
 
-      - name: Build and Deploy
-        run: mkdocs gh-deploy --force
\ No newline at end of file
+      - name: Configure Git
```
Comment (pawbu) [inline .github/workflows/docs-deploy.yml:23]: Why didn't [they](https://github.com/statisticsnorway/nais-system/blob/main/.github/workflows/publish-gh-pages.yml) have to do that?

File: .github/workflows/docs-deploy.yml [hunk 2]
```diff
@@ -10,8 +10,6 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-        with:
```
Comment (DanielElisenberg) [inline .github/workflows/docs-deploy.yml:13]: As long as you remove this like you have done I think that should be enough? (Seems like the default is true) Then the git credentials from the checkout are persisted 👀

File: .github/workflows/docs-deploy.yml (hunk 1 above)
Comment (haraldsvik) [reply .github/workflows/docs-deploy.yml:23]: Not sure, maybe they have different repo settings?

Saw multiple places saying you needed to add credentials
from the material theme:
https://squidfunk.github.io/mkdocs-material/publishing-your-site/
it was one of the things which the previous deploy action said they handled..

File: .github/workflows/docs-deploy.yml (hunk 2 above)
Comment (haraldsvik) [reply .github/workflows/docs-deploy.yml:13]: it worked 💯

**Review by DanielElisenberg — APPROVED** [review]
:100:

**Pull request merged**